server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/Payload.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.james.blob.objectstorage;
+
+import java.util.Optional;
+
+public class Payload {
+
+    private final org.jclouds.io.Payload payload;
+    private final Optional<Long> length;
+
+    public Payload(org.jclouds.io.Payload payload, Optional<Long> length) {
```

Comment:
why it's optional?

Reply:
There are two cases:
- the content is encrypted, then you have to give it's length
- the content is not encrypted, then you don't have to has you already know it's length.

Reply:
Could it be interesting to make it more explicit? Like Payload.ofEncrypted(payload, length) and Payload.ofUnencrypted(payload)?